Assets/Scripts/SafeZoneLogic.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SafeZoneLogic : MonoBehaviour
+{
+    [SerializeField]
+    private GameObject player;
```

Comment:
Please have it non serialized and get the player object from a tag "Player" in `Start()`